aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPlugin.java

```diff
@@ -93,32 +96,45 @@ public void enable() {
      * {@inheritDoc}
      */
     @Override
-    public void identifyUser(@NonNull String userId, @NonNull AnalyticsUserProfile profile) {
+    public void identifyUser(@NonNull String userId, @Nullable UserProfile userProfile) {
+        Objects.requireNonNull(userId);
         EndpointProfile endpointProfile = targetingClient.currentEndpoint();
         // Assign userId to the endpoint.
         EndpointProfileUser user = new EndpointProfileUser();
         user.setUserId(userId);
         endpointProfile.setUser(user);
         // Add user-specific data to the endpoint
-        addUserProfileToEndpoint(endpointProfile, profile);
+        addUserProfileToEndpoint(endpointProfile, userProfile);
         // update endpoint
         targetingClient.updateEndpointProfile();
     }
 
     /**
-     * Add user specific data from {@link AnalyticsUserProfile} to the endpoint profile.
+     * Add user specific data from {@link UserProfile} to the endpoint profile.
      * @param endpointProfile endpoint profile.
      * @param userProfile user specific data to be added to the endpoint.
      */
     private void addUserProfileToEndpoint(EndpointProfile endpointProfile,
-                                          AnalyticsUserProfile userProfile) {
-        endpointProfile.addAttribute("email", Arrays.asList(userProfile.getEmail()));
-        endpointProfile.addAttribute("name", Arrays.asList(userProfile.getName()));
-        endpointProfile.addAttribute("plan", Arrays.asList(userProfile.getPlan()));
+                                          UserProfile userProfile) {
+        if (userProfile.getEmail() != null) {
+            endpointProfile.addAttribute("email", Collections.singletonList(userProfile.getEmail()));
```

Comment:
Can you get these keys into an enum or StringDef, too? `PinpointProfileAttribute.EMAIL`, etc.? You could add a little helper method like:
```
private void addAttribute(EndpointProfile profile, @PinpointProfileAttribute String profileAttribute, String value) {
    profile.addAttribute(profileAttribute, Collections.singletonList(value));
}
```

^^ Above, the at-PinpointProfileAttribute is a StringDef